outbound/state/work_description.py

```diff
@@ -0,0 +1,132 @@
+            raise ValueError('Expected persistence store')
+
+        self.persistence_store = persistence_store
+        self._deserialize_data(store_data)
+
+    async def publish(self):
+        """
+        Attempts to publish the local state of the work description to the state store
+        :return:
+        """
+        logger.info(f'Attempting to publish work description {self.operation_id}')
+
+        serialised = self._serialise_data()
+
+        await self.persistence_store.add(self.operation_id, serialised)
+        logger.info(f'Successfully updated work description to state store for {self.operation_id}')
+
+    def _deserialize_data(self, store_data):
+        data_attribute = store_data['DATA']
+        self.operation_id: str = store_data.get('OPERATION_ID')
+        self.transaction_id: int = data_attribute.get('TRANSACTION_ID')
+        self.transaction_timestamp: str = data_attribute.get('TRANSACTION_TIMESTAMP')
+        self.transaction_type: str = data_attribute.get('TRANSACTION_TYPE')
+        self.sis_sequence: int = data_attribute.get('SIS_SEQUENCE')
+        self.sms_sequences: list = data_attribute.get('SMS_SEQUENCES')
+        self.sender: str = data_attribute.get('SENDER')
+        self.recipient: str = data_attribute.get('RECIPIENT')
+
+    def _serialise_data(self):
+        """
+        A simple serialization method that produces an object from the local data which can be stored in the
+        persistence store
+        """
+        return {
+            OPERATION_ID: self.operation_id,
+            DATA: {
+                TRANSACTION_ID: self.transaction_id,
+                TRANSACTION_TIMESTAMP: self.transaction_timestamp,
+                TRANSACTION_TYPE: self.transaction_type,
+                SIS_SEQUENCE: self.sis_sequence,
+                SMS_SEQUENCES: self.sms_sequences,
+                SENDER: self.sender,
+                RECIPIENT: self.recipient
+            }
+        }
+
+
+def create_new_work_description(persistence_store: pa.PersistenceAdaptor,
+                                segments
+                                ) -> WorkDescription:
+    """
+    Builds a new local work description instance given the details of the message, these details are held locally
+    until a `publish` is executed
+    """
+
+    operation_id = get_uuid()
+    sms_sequences = []
+
+    for segment in segments:
+        if isinstance(segment, InterchangeHeader):
+            transaction_timestamp = segment.date_time
+            sender = segment.sender
+            recipient = segment.recipient
+        if isinstance(segment, (InterchangeHeader, InterchangeTrailer)):
+            sis_sequence = segment.sequence_number
+        if isinstance(segment, (MessageHeader, MessageTrailer)):
+            sms_sequences.append(segment.sequence_number)
+        if isinstance(segment, ReferenceTransactionNumber):
+            transaction_id = segment.reference
+        if isinstance(segment, ReferenceTransactionType):
```

Comment:
It would be slightly quicker to use if...elif for these to avoid doing all of the checks every time. Could you make this change to the converter as well?